test/happypath.test.js

```diff
@@ -101,6 +105,31 @@ describe('Complete flow test', function () {
         expect(project.uuid).to.equal(projUuid)
     })
 
+    it.only('Must be able to update and send correct mail', async () => {
+        const content = '<p>New mail confirmation email: <a href="{{& link}}">{{& link}}</a></p>'
+        const title = 'New mail confirmation title'
+        const mailType = 'MAIL_CONFIRMATION_MAIL'
+        const lang = 'EN'
+
+        // Create Admin
+        let admin = await TestUser.createAdmin('admin@email.com')
+        await db.insertUser(admin)
+        await admin.getJwtToken()
+
+        // Admin updates mail confirmation mail
+        let updatedDepositRequestMail = await cmsSvc.updateMail(admin, db.COOP, mailType, lang, content, title)
+        let depositRequestMailList = await cmsSvc.getMail(db.COOP, mailType, lang)
+        expect(depositRequestMailList.mails).to.have.length(1)
+        expect(updatedDepositRequestMail).to.be.deep.equal(depositRequestMailList.mails[0])
+
+        // Admin requests confirmation mail resend
+        await userSvc.resendMailConfirmation(admin)
+
+        let mailList = await retry(fakeStmpSvc.getMails, 100)
+        expect(mailList).to.have.length(1)
+        expect(mailList[0].subject).to.be.equal(title)
```

Comment:
Is it possible to assert mail body has the correct text?

Reply:
The exact mail text is created in the mail service, so that's the internal implementation of the mail service. In this case, the link is set in the LinkResolver class. That is not exposed outside, I don't see any value in testing that, we already do it inside mail service tests.

Reply:
Ok, do we have some assurance that the mail text is in the new format and not in the default?

Reply:
suggestion ```
expect(updatedDepositRequestMail).to.be.deep.equal(depositRequestMailList.mails[0])
```
We have the assurance it is updated inside cms service

Reply:
```suggestion
        expect(mailList[0].subject).to.be.equal(title)
```

We have assurance it is updated inside cms service

Reply:
```suggestion
        expect(updatedDepositRequestMail).to.be.deep.equal(depositRequestMailList.mails[0])
```

We have assurance it is updated inside cms service

Reply:
If it's vital I can add the expectation for the mail text as well

Reply:
I know that template is updated in cms but the problem is if the mail-service does not fetch the new template and use the default. You don't have to assert the whole text is in the correct form, just it is not in default form or it contains some text specific to the new template.

Reply:
Sure, I'll do that.